03/runoff/runoff.c

```diff
@@ -0,0 +1,204 @@
+            }
+        }
+
+        printf("\n");
+    }
+
+    // Keep holding runoffs until winner exists
+    while (true)
+    {
+        // Calculate votes given remaining candidates
+        tabulate();
+
+        // Check if election has been won
+        bool won = print_winner();
+        if (won)
+        {
+            break;
+        }
+
+        // Eliminate last-place candidates
+        int min = find_min();
+        bool tie = is_tie(min);
+
+        // If tie, everyone wins
+        if (tie)
+        {
+            for (int i = 0; i < candidate_count; i++)
+            {
+                if (!candidates[i].eliminated)
+                {
+                    printf("%s\n", candidates[i].name);
+                }
+            }
+            break;
+        }
+
+        // Eliminate anyone with minimum number of votes
+        eliminate(min);
+
+        // Reset vote counts back to zero
+        for (int i = 0; i < candidate_count; i++)
+        {
+            candidates[i].votes = 0;
+        }
+    }
+    return 0;
+}
+
+// Record preference if vote is valid
+bool vote(int voter, int rank, string name)
+{
+    for(int i = 0; i < candidate_count; i++) {
+        if(strcmp(name, candidates[i].name) == 0) {
+            preferences[voter][rank] = i;
+            return true;
+        }
+    }
+    return false;
+}
+
+// Tabulate votes for non-eliminated candidates
+void tabulate(void)
+{
+    int preferredCandidate;
+
+    for(int i = 0; i < voter_count; i++) {
+
+        for(int j = 0; j < candidate_count; j++) {
+
+            preferredCandidate = preferences[i][j];
+
+            if(!candidates[preferredCandidate].eliminated) {
+                candidates[preferredCandidate].votes++;
+                break;
+            }
+        }
+    }
+    return;
+}
+
+// Print the winner of the election, if there is one
+bool print_winner(void)
+{
+    for(int i = 0; i < candidate_count; i++) {
+
+        if(candidates[i].votes > (voter_count / 2)) {
+            printf("Winner: %s\n", candidates[i].name);
+            return true;
+        }
+    }
+    return false;
+}
+
+// Return the minimum number of votes any remaining candidate has
+int find_min(void)
+{
+    int lowestVotes = 0;
+
+    for(int i = 0; i < candidate_count; i++) {
+        if(!candidates[i].eliminated && candidates[i].votes < lowestVotes) {
+            lowestVotes = candidates[i].votes;
+        }
+    }
+    return lowestVotes;
+}
+
+// Return true if the election is tied between all candidates, false otherwise
+bool is_tie(int min)
+{
+    for(int i = 0; i < candidate_count; i++) {
+        if(!candidates[i].eliminated && candidates[i].votes == min){
+            return true;
+        }
+    }
+    return false;
+}
+
+// Eliminate the candidate (or candidates) in last place
+void eliminate(int min)
+{
+    for(int i = 0; i < candidate_count; i++) {
+        if(candidates[i].votes == min) {
+            candidates[i].eliminated = true;
+        }
+    }
+    return;
```

Comment:
```suggestion
    for(int i = 0; i < candidate_count; i++) {
        if(candidates[i].votes == min) candidates[i].eliminated = true;
    }
    
    return;
```
s.o. + Vereinfachung + Leere Zeile nach der Schleife/vor return